tests/acceptance/ContactManagementCest.php

```diff
@@ -0,0 +1,118 @@
+        $I->amOnPage('/s/contacts');
+
+        // Wait for and click on the "Quick Add" button
+        $I->waitForElementClickable('#toolbar .quickadd', 30);
+        $I->click('#toolbar .quickadd');
+
+        // Wait for the Quick Add modal to appear and verify its presence
+        $I->waitForElementVisible('#MauticSharedModal-label', 30);
+        $I->see('Quick Add', 'h4.modal-title');
+
+        // Wait for the form fields to be visible
+        $I->waitForElementVisible('#lead_firstname', 10);
+
+        // Fill out the quick add form
+        $I->fillField('#lead_firstname', 'QuickAddFirstName');
+        $I->fillField('#lead_lastname', 'QuickAddLastName');
+        $I->fillField('#lead_email', 'quickadd@example.com');
+        $I->fillField('#lead_tags_chosen input', 'TestTag');
+        $I->pressKey('#lead_tags_chosen input', \Facebook\WebDriver\WebDriverKeys::ENTER);
+        $I->fillField('#lead_companies_chosen input', 'Mautic');
+        $I->pressKey('#lead_companies_chosen input', \Facebook\WebDriver\WebDriverKeys::ENTER);
+
+        // Wait for the save button to be clickable and submit the form
+        $I->waitForElementClickable('#MauticSharedModal > div > div > div.modal-footer > div > button.btn.btn-default.btn-save.btn-copy', 30);
+        $I->click('#MauticSharedModal > div > div > div.modal-footer > div > button.btn.btn-default.btn-save.btn-copy');
+        $I->waitForElementNotVisible('#MauticSharedModal-label', 30);
+
+        $I->amOnPage('/s/contacts');
+        $I->reloadPage(); // Ensure the latest data is loaded
+
+        // Search for the contact we just created
+        $I->fillField('#list-search', 'QuickAddFirstName');
+        $I->pressKey('#list-search', \Facebook\WebDriver\WebDriverKeys::ENTER);
+        $I->waitForElementVisible('#leadTable', 10); // Wait for the search results to appear
+
+        // Verify the contact is in the list
+        $I->see('QuickAddFirstName', '#leadTable');
+
+        // Clear the search
+        $I->click('#btn-filter');
+    }
+
+    public function createContactFromForm(AcceptanceTester $I)
+    {
+        // Navigate to the contacts page
+        $I->amOnPage('/s/contacts');
+
+        // Click on "+New" button
+        $I->waitForElementClickable('#toolbar a:nth-child(2)', 30);
+        $I->click('#toolbar a:nth-child(2)');
+        $I->waitForText('New Contact', 30);
+
+        // Fill out the form fields
+        $I->waitForElementVisible('#lead_firstname', 10);
+        $I->fillField('#lead_firstname', 'FirstName');
+        $I->fillField('#lead_lastname', 'LastName');
+        $I->fillField('#lead_email', 'email@example.com');
+
+        // Fill Company field
+        // $I->click('#lead_companies_chosen');
+        // $I->fillField('#lead_companies_chosen > ul > li > input', 'TestCompany');
+        // $I->wait(2);
+        // $I->see('No matches found TestCompany');
+        // $I->fillField('#lead_companies_chosen > ul > li > input', 'Create');
```

Comment:
Please don't commit to the comment code.

Reply:
Okay, I'll keep it in mind.